dbt-tests-adapter/dbt/tests/adapter/incremental/test_incremental_microbatch.py

```diff
@@ -3,9 +3,13 @@
 from unittest import mock
 
 import pytest
-from freezegun import freeze_time
 
-from dbt.tests.util import relation_from_name, run_dbt
+try:
+    # patch_microbatch_end_time introduced in dbt 1.9.0
+    from dbt.tests.util import relation_from_name, run_dbt, patch_microbatch_end_time
+except ImportError:
+    from dbt.tests.util import relation_from_name, run_dbt
+    from freezegun import freeze_time as patch_microbatch_end_time
```

Comment:
Since `relation_from_name` and `run_dbt` are shared and are not expected to raise `ImportError`, I would move them outside of the try block.

```suggestion
from dbt.tests.util import relation_from_name, run_dbt
try:
    # patch_microbatch_end_time introduced in dbt 1.9.0
    from dbt.tests.util import patch_microbatch_end_time
except ImportError:
    from freezegun import freeze_time as patch_microbatch_end_time
```

Reply:
This makes sense, less duplicate code, only try/excepting around thing that worried about failing 👍 I'll get it changed